src/_P042_Candle.ino

```diff
@@ -76,422 +83,440 @@ enum ColorType {
+      addNumericBox(F("wBlue"),  PCONFIG(2), 0, 255);
+
+      // Brightness Selection
+      addRowLabel(F("Brightness"));
+      addHtml(F("min<input type='range' id='brSlide' min='0' max='255' value='"));
+      addHtmlInt(PCONFIG(3));
+      addHtml(F("'> max"));
 
-    case PLUGIN_WEBFORM_SAVE:
       {
-        PCONFIG(0) = getFormItemInt(F("web_RGB_Red"));
-        PCONFIG(1) = getFormItemInt(F("web_RGB_Green"));
-        PCONFIG(2) = getFormItemInt(F("web_RGB_Blue"));
-        PCONFIG(3) = getFormItemInt(F("web_Bright_Text"));
-        PCONFIG(4) = getFormItemInt(F("web_Candle_Type"));
-        PCONFIG(5) = getFormItemInt(F("web_Color_Type"));
-
-        Candle_red = PCONFIG(0);
-        Candle_green = PCONFIG(1);
-        Candle_blue = PCONFIG(2);
-        if (Candle_bright > 255) {
-          Candle_bright = 255;
-        }
-        Candle_bright = PCONFIG(3);
-        Candle_type = (SimType)PCONFIG(4);
-        Candle_color = (ColorType)PCONFIG(5);
+        // char tmpString[128];
+        // sprintf_P(tmpString, PSTR("<input type='text' name='brText' id='brText' size='3' value='%u'>"), PCONFIG(3));
+        // addHtml(concat(F("<input type='text' name='brText' id='brText' size='3' value='"),) + F("'>"));
+        addFormNumericBox(F("Brightness Value"), F("brText"), PCONFIG(3), 0, 255);
+      }
 
-        success = true;
-        break;
+      // Some Javascript we need to update the items
+      addHtml(F("<script script type='text/javascript'>"));
+
+      // "function update(picker) {"
+      // "    document.getElementById('wRed').value = Math.round(picker.rgb[0]);"
+      // "    document.getElementById('wGreen').value = Math.round(picker.rgb[1]);"
+      // "    document.getElementById('wBlue').value = Math.round(picker.rgb[2]);"
+      // '}'
+      // Minified:
+      addHtml(F(
+                "function update(e){document.getElementById('wRed').value=Math.round(e.channel('R')),document.getElementById('wGreen').value=Math.round(e.channel('G')),document.getElementById('wBlue').value=Math.round(e.channel('B'))}"));
+
+      // "window.addEventListener('load', function(){"
+      // "var slider = document.getElementById('brSlide');"
+      // "slider.addEventListener('change', function(){"
+      // "document.getElementById('brText').value = this.value;"
+      // "});"
+      // "});</script>"
+      // Minified:
+      addHtml(F(
+                "window.addEventListener('load',function(){document.getElementById('brSlide').addEventListener('input',function(){document.getElementById('brText').value=this.value})});"));
+      addHtml(F("</script>"));
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      PCONFIG(0) = getFormItemInt(F("wRed"));
+      PCONFIG(1) = getFormItemInt(F("wGreen"));
+      PCONFIG(2) = getFormItemInt(F("wBlue"));
+      PCONFIG(3) = getFormItemInt(F("brText"));
+      PCONFIG(4) = getFormItemInt(F("cType"));
+      PCONFIG(5) = getFormItemInt(F("clrType"));
```

Comment:
Not sure how much it may save, but you could make an array of flash strings and then for-loop over them. The PCONFIG nrs are all in a sequence.

Reply:
The gain by _only_ changing that was ~80 bytes smaller .bin on an ESP8266 build.